### ISSUE TITLE
Enhancement in update_or_create method

### ISSUE THREAD

**Issue by MojixCoder** [collerek/ormar#612]
I want to change `update_or_create` method just like we did in `get_or_create`[554](https://github.com/collerek/ormar/issues/554).
Here is the Django API for this method:
```python
obj, created = Person.objects.update_or_create(
    first_name='John', last_name='Lennon',
    defaults={'first_name': 'Bob'},
)
```
I will change `defaults` to `_defaults`.

**Comment by ooliver1** [comment]
IMO the current docstring for `update_or_create` is misleading.

> Updates the model, **or in case there is no match in database** creates a new one.

It does not do this at all, it simply checks if `pk_name` has been passed, and creates a new one if not. This currently seems rather useless at the moment, since it requires an autoincrement key which you already knowingly do or don't pass.

**Comment by arlyon** [comment]
Yeah, I am finding it very strange behaviour that `update_or_create` would throw a `NoMatch` error.

**Comment by collerek** [comment]
Fixed on master in #1666 — `update_or_create` now creates a new row when a pk is supplied but no matching row exists, instead of raising `NoMatch`. Not released yet, will be in the next release.
